src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -656,10 +656,51 @@ def update_period(self):
             if exit_code > 0:
                 raise DashboardException(e=err, msg='Unable to update period',
-                                         http_status_code=500, component='rgw')
+                                         http_status_code=400, component='rgw')
         except SubprocessError as error:
             raise DashboardException(error, http_status_code=500, component='rgw')
 
+    def edit_realm(self, realm_name: str, default: bool, new_realm_name: str):
+        rgw_realm_edit_cmd = []
+        cmd_edit_realm_options = []
+        if new_realm_name == realm_name:
+            if str_to_bool(default):
+                rgw_realm_edit_cmd = ['realm', 'default']
+                cmd_edit_realm_options = ['--rgw-realm', realm_name]
+                rgw_realm_edit_cmd += cmd_edit_realm_options
+                try:
+                    exit_code, _, err = mgr.send_rgwadmin_command(rgw_realm_edit_cmd)
+                    if exit_code > 0:
+                        raise DashboardException(e=err, msg='Unable to set {} as default realm'.format(realm_name),  # noqa E501  #pylint: disable=line-too-long
+                                                 http_status_code=500, component='rgw')
+                except SubprocessError as error:
+                    raise DashboardException(error, http_status_code=500, component='rgw')
+            else:
+                raise DashboardException(msg='The realm already exists',
+                                         http_status_code=400, component='rgw')
+        else:
+            rgw_realm_edit_cmd = ['realm', 'rename']
+            cmd_edit_realm_options = ['--rgw-realm', realm_name, '--realm-new-name', new_realm_name]
+            rgw_realm_edit_cmd += cmd_edit_realm_options
+            try:
+                exit_code, _, err = mgr.send_rgwadmin_command(rgw_realm_edit_cmd, False)
+                if exit_code > 0:
+                    raise DashboardException(e=err, msg='Unable to edit realm',
+                                             http_status_code=500, component='rgw')
+            except SubprocessError as error:
+                raise DashboardException(error, http_status_code=500, component='rgw')
+            if str_to_bool(default):
+                rgw_realm_edit_cmd = ['realm', 'default']
+                cmd_edit_realm_options = ['--rgw-realm', new_realm_name]
+                rgw_realm_edit_cmd += cmd_edit_realm_options
+                try:
+                    exit_code, _, _ = mgr.send_rgwadmin_command(rgw_realm_edit_cmd, False)
+                    if exit_code > 0:
+                        raise DashboardException(msg='Unable to set {} as default realm'.format(new_realm_name),  # noqa E501  #pylint: disable=line-too-long
+                                                 http_status_code=500, component='rgw')
+                except SubprocessError as error:
+                    raise DashboardException(error, http_status_code=500, component='rgw')
```

Comment:
Here I see a logic problem. This whole method's logic can be simplified by reordering operations. Right now we have.

```python
if name_different && default:
  set default
  
else:
  rename
  set default
```
That also makes me think to whether you can toggle default. Can't you move from default being true to false? Here it says it can only be enabled but not disabled (haven't checked if this how rgw realms work but I would be surprised).

So... should be able to simply to something like:
```python
if default is different:
  set new default
if rename
  rename
```

Reply:
@pereman2 , If you have set one entity as default, you cannot set default as false on that entity, you can simply mark the other entity as default and it will automatically set default false for the first one. You can check the docs here - https://docs.ceph.com/en/latest/radosgw/multisite/

Reply:
this wasn't addressed, you are repeating code in the else block without being necessary

Reply:
@pereman2 , I didn't get your point here, where is the code being repeated?- 
1. If realm is not default and the user sets it to default on editing, we are doing it in the first step.
2. If the user re-names the realm and set it as default that was not default at first, we rename the realm first and then make it as default.
Is there anything that am missing here?

Reply:
I mean, that this logic can be simplified then to:
```python
def edit(default, new_name):
  if new_name != name:
    send rename command
  if default:
    send default command
 ```

Reply:
It covers both paths without having to over nest things

Reply:
Fixed it. Thanks @pereman2